gtest/src/manager.rs

```diff
@@ -422,6 +427,13 @@ impl ExtManager {
 
         core_processor::handle_journal(journal, self);
     }
+
+    pub(crate) fn take_message(&mut self, program_id: &ProgramId, index: usize) -> StoredMessage {
+        self.actor_to_mailbox
+            .get_mut(program_id)
+            .expect("No mailbox with such program id")
+            .remove(index)
+    }
```

Comment:
This is useless logic for manager. I would recommend do it manually in mailbox methods